dcos/nodeutil/util.go

```diff
@@ -121,7 +124,10 @@ func NewNodeInfo(client *http.Client, role string, options ...Option) (NodeInfo,
 		detectIPTimeout:   defaultExecTimeout,
 		dnsRecordLeader:   dcos.DNSRecordLeader,
 		mesosStateURL:     defaultStateURL.String(),
-		clusterIDLocation: defaultClusterIDLocation,
+		clusterIDLocation: defaultClusterIDLocationLinux,
+	}
+	if runtime.GOOS == "windows" {
```

Comment:
Here and elsewhere when we overwrite a variable based on the runtime, I'd prefer to just put these things into functions.  So, something like:

```
func getClusterIDLocation() string {
  switch runtime.GOOS {
    case "windows":
      return "\\mesos\\var\\lib\\dcos\\cluster-id"
    default:
      return "/var/lib/dcos/cluster-id
  }
}
```

And then we can initialize the struct like this:

``` 
d := &dcosInfo{
  ....
  clusterIDLocation: getClusterIDLocation(),
}
	
```

This makes the code that calls it easier to read, and we avoid having to declare runtime-specific constants like `defaultClusterIDLocationLinux` and `defaultClusterIDLocationWindows`.